fdbrpc/sim2.actor.cpp

```diff
@@ -1681,6 +1681,25 @@ class Sim2 final : public ISimulator, public INetworkConnections {
 		}
 		return result;
 	}
+	bool killDataHall(Optional<Standalone<StringRef>> dataHallId,
+	                  KillType kt,
+	                  bool forceKill,
+	                  KillType* ktFinal) override {
+		auto processes = getAllProcesses();
+		std::set<Optional<Standalone<StringRef>>> dataHallMachines;
+		for (auto& process : processes) {
+			if (process->locality.dataHallId() == dataHallId) {
+				dataHallMachines.insert(process->locality.machineId());
+			}
+		}
+		bool result = false;
+		for (auto& machineId : dataHallMachines) {
+			if (killMachine(machineId, kt, forceKill, ktFinal)) {
+				result = true;
+			}
+		}
```

Comment:
Why `result = any(killMachine(dataHallMachines[i]))` but not `result=all(killMachine(dataHallMachines[i]))`?

Reply:
That is the same logic as killZone (and killMachine). They return `true' if they didn't fully succeed in performing the kill. I don't know why that's useful, and the return value is mostly ignored by callers. I just followed precedent here.